src/main/java/fi/pohina/vinkkilista/domain/Blog.java

```diff
@@ -0,0 +1,75 @@
+package fi.pohina.vinkkilista.domain;
+
+import java.util.*;
+
+public class Blog implements Bookmark {
+
+    private String title;
+    private String author;
+    private String url;
+    private BookmarkType type;
+    private ArrayList<Course> relatedCourses;
+
+    /**
+     * Constructor
+     * @param title
+     * @param url
+     * @param type 
+     */
+    public Blog(String title, String url, BookmarkType type) {
+        this.title = title;
+        this.url = url;
+        this.type = type;
+
+        relatedCourses = new ArrayList<Course>();
+    }
+    
+    /**
+     * Constructor
+     * @param title
+     * @param url
+     * @param type
+     * @param author
+     */
+    public Blog(String title, String url, BookmarkType type, String author) {
+        this.title = title;
+        this.url = url;
+        this.type = type;
+        this.author = author;
+        
+        relatedCourses = new ArrayList<Course>();
+    }
```

Comment:
Could re-use the other constructor:

```java
public Blog(String title, String url, BookmarkType type, String author) {
    this(title, url, type); // <-- calls the other constructor, allows us to re-use the initialization logic
    this.author = author;
}
```

with `BookmarkType` removed since the getter can just return the correct type altogether (remove the unused field too maybe?):

```java
public Blog(String title, String url) {
    this.title = title;
    this.url = url;
    relatedCourses = new ArrayList<Course>();
}

public Blog(String title, String url, String author) {
    this(title, url); // <-- calls the other constructor, allows us to re-use the initialization logic
    this.author = author;
}
```